File: dev-docs/snippets/group-type-configs/create.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Create a new group type configuration. `groupType` is required; `ruleSetId`
// and `autoAddCreator` (defaults to false) are optional.
export async function create(client: JsBaoClient, ruleSetId: string) {
  // #region example
  const config = await client.groupTypeConfigs.create({
    groupType: "team",
    ruleSetId,
    autoAddCreator: true,
  });
  // #endregion example
  return config;
}

File: dev-docs/snippets/group-type-configs/create.swift
import JsBaoClient

// Create a new group type configuration. Swift takes an untyped `[String: Any]`
// params dictionary in place of `CreateGroupTypeConfigParams`.
func create(client: JsBaoClient, ruleSetId: String) async throws {
  // #region example
  let config = try await client.groupTypeConfigs.create(params: [
    "groupType": "team",
    "ruleSetId": ruleSetId,
    "autoAddCreator": true,
  ])
  // #endregion example
  _ = config
}
